batch/batch/server/server.py

```diff
@@ -490,29 +490,32 @@ def is_successful(self):
                 log.warning(f'job {self.id} has pod {pod.metadata.name} which is '
                             f'terminated but has no timing information. {pod}')
                 self.duration = None
-            await db.jobs.update_record(self.id,
-                                        exit_code=self.exit_code,
-                                        duration=self.duration,
-                                        pod_name=None)
+            await db.jobs.update_record(self.id, **{db.jobs.exit_code_field(task_name): exit_code,
+                                                    'duration': self.duration,
+                                                    'pod_name': None})
```

Comment:
So I think we should always update exit_code and log together.  So let's join this database update with the one inside `_write_log`.

I think we should increment the index when we write the exit_code, otherwise it is temporarily inconsistent.